credentials/credentials.go

```diff
@@ -215,6 +215,21 @@ func RequestInfoFromContext(ctx context.Context) (ri RequestInfo, ok bool) {
 	return ri, ok
 }
 
+// NewRequestInfoContext creates a new context with the given RequestInfo
+// attached to it.
```

Comment:
`NewContextWithRequestInfo attaches the RequestInfo to ctx`?

Reply:
I chose "creates a new context with ... attached" because many other existing functions are documented that way:

```
// NewIncomingContext creates a new context with incoming md attached.
// NewOutgoingContext creates a new context with outgoing md attached.
// NewContext creates a new context with peer information attached.
// NewContextWithServerTransportStream creates a new context from ctx and attaches stream to it.
```

Reply:
Of these, the last one is most accurate.  I'm also fine with `creates a new context from ctx and attaches ri to it`, too.

Reply:
Changed it to `NewContextWithRequestInfo creates a new context from ctx and attaches ri to it.`